board/rriv_board_0_4_2/src/lib.rs

```diff
@@ -53,57 +69,107 @@ pub struct Serial {
+        // let scl = gpiob.pb6.into_alternate_open_drain(&mut gpiob.crl); // i2c
+        // let sda = gpiob.pb7.into_alternate_open_drain(&mut gpiob.crl); // i2c
+        // let mut switched_pow = gpioc.pc6.into_push_pull_output(&mut gpioc.crl);
+        // let mut hse_pin = gpioc.pc13.into_push_pull_output(&mut gpioc.crh); 
+        
+
+// type aliases to make things tenable
+type BoardI2c = BlockingI2c<
+    I2C1,
+    (
+        Pin<'B', 6, Alternate<OpenDrain>>,
+        Pin<'B', 7, Alternate<OpenDrain>>,
+    ),
+>;
+
 pub struct Board {
-    pub serial: Option<Serial>,
+    pub i2c1: Option<BoardI2c>,
+    pub delay: Option<SysDelay>,
+    pub pins: Option<Pins>
 }
 
 impl Board {
     pub fn new() -> Self {
-        Board {
-            serial: None,
-        }
+        Board { i2c1: None, delay: None, pins: None}
     }
 }
 
+
 impl RRIVBoard for Board {
     fn setup(&mut self) {
         rprintln!("board new");
 
+        let mut core_peripherals = cortex_m::Peripherals::take().unwrap();
+
         // Get access to the device specific peripherals from the peripheral access crate
-        let p = pac::Peripherals::take().unwrap();
+        let device_peripherals = pac::Peripherals::take().unwrap();
+        // rprintln!("{:?}",device_peripherals.RCC.apb2enr); //.i2c1en();  //enabled();
+        // device_peripherals.RCC.apb2enr.
 
         // Take ownership over the raw flash and rcc devices and convert them into the corresponding
         // HAL structs
-        let mut flash = p.FLASH.constrain();
-        let rcc = p.RCC.constrain();
+        let mut flash = device_peripherals.FLASH.constrain();
+        let rcc = device_peripherals.RCC.constrain();
+        let mut afio = device_peripherals.AFIO.constrain();     // Prepare the alternate function I/O registers
+
+
+        // Prepare the GPIO
+        let mut gpioa = device_peripherals.GPIOA.split();
+        let mut gpiob = device_peripherals.GPIOB.split();
+        let mut gpioc = device_peripherals.GPIOC.split();
+        let mut gpiod = device_peripherals.GPIOD.split();
+
+
+        // Set up pins
+        let mut pins = Pins::build(gpioa, gpiob, gpioc, gpiod);
+        // self.pins = Some(pins);
+        // let pins = self.pins;
+
+ 
+        let mut switched_pow = pins.enable_3v.into_push_pull_output(&mut pins.gpioc.crl);
+        let mut hse_pin = pins.enable_hse.into_push_pull_output(&mut pins.gpioc.crh); // high speed external on/off
+        
+        let wake = pins.wake_button();
+
+        // Turn on the HSE
+        hse_pin.set_high();
 
         // Freeze the configuration of all the clocks in the system
         // and store the frozen frequencies in `clocks`
         let clocks = rcc
-        .cfgr
-        .use_hse(8.MHz())
-        .sysclk(48.MHz())
-        .pclk1(24.MHz())
-        .freeze(&mut flash.acr);
+            .cfgr
+            .use_hse(8.MHz())
+            .sysclk(48.MHz())
+            .pclk1(24.MHz()) // was 24, set to 6 following i2c code
```

Comment:
💭 I thought I changed pclk1 to 24 for usb specifically. did happen you test the usb functionality with this change?